### PR TITLE
Statistics reporting via Rest API

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

File: src/python/foglamp/core/statistics_db_services.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    results = {}
+    for row in rows:
+        temp = OrderedDict(zip(columns, row))
+        results.update({temp['key']: temp['value']})
+
+    await conn.close()
+
+    return results
+
+async def read_statistics_history(limit):
```
Comment (praveen-garg) [inline src/python/foglamp/core/statistics_db_services.py]: limit=None

File: src/python/foglamp/core/statistics_db_services.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+async def read_statistics_history(limit=None):
+    conn = await asyncpg.connect(database=__DB_NAME)
+    _limit_clause = " LIMIT $1" if limit else " "
+
+    query = """
+                SELECT date_trunc('second', history_ts::timestamptz)::varchar as history_ts,
+                        key,
+                        value FROM statistics_history
+                WHERE history_ts IN (SELECT distinct history_ts FROM statistics_history ORDER BY history_ts DESC {limit_clause})
```
Comment (ashwinscale) [inline src/python/foglamp/core/statistics_db_services.py]: This only works if history_ts is the same for all statistics in a single grouping... which it is not currently.

**Comment by amarendra-dianomic** [comment]
@ashwinscale PR rebased. Please check.

File: src/python/foglamp/core/api.py [hunk 3]
```diff
@@ -36,6 +36,8 @@
     | POST            | /foglamp/task                                             |
     | GET DELETE      | /foglamp/task/{task_id}                                   |
 
+    | GET             | /foglamp/statistics                                        |
+    | GET             | /foglamp/statistics/history                                |
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/api.py]: Please delete one space before "|"

File: src/python/foglamp/core/api.py [hunk 4]
```diff
@@ -275,3 +277,44 @@
...
+        return web.json_response(statistics)
+    except Exception as ex:
+        raise web.HTTPInternalServerError(reason='FogLAMP has encountered an internal error', text=str(ex))
+
+
+async def get_statistics_history(request):
+    """
+        Returns a list of general set of statistics
+
+        Example: Curl -X GET -d limit=1 http://localhost:8082/foglamp/statistics/history
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/api.py]: There is no command ~Curl~, use **curl**

File: src/python/foglamp/core/api.py [hunk 5]
```diff
@@ -275,3 +277,44 @@
...
+    """
+
+    try:
+        limit = request.query.get('limit') if 'limit' in request.query else 0
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/api.py]: Its better to add __DEFAULT_LIMIT=?

File: src/python/foglamp/core/statistics_db_services.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                        key,
+                        value FROM statistics_history
+                WHERE date_trunc('second', history_ts::timestamptz)::varchar IN (SELECT distinct date_trunc('second', history_ts::timestamptz)::varchar as history_ts FROM statistics_history ORDER BY history_ts DESC {limit_clause})
+                ORDER BY history_ts, key;
+            """.format(limit_clause=_limit_clause)
+
+    stmt = await conn.prepare(query)
+    rows = await stmt.fetch(limit) if limit else await stmt.fetch()
+
+    columns = ('history_ts', 'key', 'value')
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/core/statistics_db_services.py]: Do we need "history_ts" return as JSON object? Not as per doc

Comment (amarendra-dianomic) [inline src/python/foglamp/core/statistics_db_services.py]: added as an aid to debugging/verification.

File: src/python/foglamp/core/api.py (hunk 5 above)
Comment (amarendra-dianomic) [inline src/python/foglamp/core/api.py]: limit is totally optional. So not needed.

**Comment by amarendra-dianomic** [comment]
@ashish-ScaleDB all required changes done. Pl check.

**Pull request merged**